packages/serverpod/lib/src/server/health_check.dart

```diff
@@ -51,11 +51,13 @@ Future<ServerHealthResult> defaultHealthCheckMetrics(
     );
 
     var session = await pod.createSession(enableLogging: false);
-    await databaseConnection.insert(entry, session: session);
+    await databaseConnection.legacy.insert(entry, session: session);
```

Comment:
What is the strategy for the legacy connection here?

Reply:
This looks like a mistake on my part! Fixed!